model/event/host_event_finder.go

```diff
@@ -0,0 +1,62 @@
+package event
+
+import (
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/mongodb/anser/bsonutil"
+	"gopkg.in/mgo.v2/bson"
+)
+
+func getRecentStatusesForHost(hostId string, n int) (int, []string) {
+	pipeline := []bson.M{
+		{"$match": bson.M{
+			TypeKey:       EventTaskFinished,
+			ResourceIdKey: hostId,
+			bsonutil.GetDottedKeyName(DataKey, hostDataResourceTypeKey): ResourceTypeHost,
+		}},
+		{"$sort": bson.M{TimestampKey: -1}},
+		{"$limit": n},
+		{"$group": bson.M{
+			"_id":    nil,
+			"count":  bson.M{"$sum": 1},
+			"status": bson.M{"$addToSet": "$" + bsonutil.GetDottedKeyName(DataKey, hostDataStatusKey)}}},
```

Comment:
I think this aggregation is trying to get the count of statuses for this host for the past n events, but won't this pipeline stage just return all the statuses that have been seen in the past n events (not their counts)?